cwms-data-api/src/main/java/cwms/cda/api/MeasurementController.java

```diff
@@ -0,0 +1,247 @@
+
+public final class MeasurementController implements CrudHandler {
+
+    static final String TAG = "Measurements";
+
+    private final MetricRegistry metrics;
+    private final Histogram requestResultSize;
+
+    public MeasurementController(MetricRegistry metrics) {
+        this.metrics = metrics;
+        String className = this.getClass().getName();
+        requestResultSize = this.metrics.histogram(name(className, "results", "size"));
+    }
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = OFFICE_MASK, description = "Office id for filtering measurements."),
+                    @OpenApiParam(name = ID_MASK, description = "Location id for filtering measurements."),
+                    @OpenApiParam(name = MIN_NUMBER, description = "Minimum number for filtering measurements."),
+                    @OpenApiParam(name = MAX_NUMBER, description = "Maximum number for filtering measurements."),
+                    @OpenApiParam(name = MIN_DATE, description = "Minimum date for filtering measurements."),
+                    @OpenApiParam(name = MAX_DATE, description = "Maximum date for filtering measurements."),
+                    @OpenApiParam(name = UNIT_SYSTEM, description = "Specifies the unit system"
+                            + " of the response. Valid values for the unit field are: "
+                            + "\n* `EN`  Specifies English unit system.  Location values will be in the "
+                            + "default English units for their parameters."
+                            + "\n* `SI`  Specifies the SI unit system.  Location values will be in the "
+                            + "default SI units for their parameters. If not specified, EN is used.")
+            },
+            responses = {
+                    @OpenApiResponse(status = "200", content = {
+                            @OpenApiContent(isArray = true, type = Formats.JSONV1, from = Measurement.class),
+                            @OpenApiContent(isArray = true, type = Formats.JSON, from = Measurement.class)
+                    })
+            },
+            description = "Returns matching measurement data.",
+            tags = {TAG}
+    )
+    @Override
+    public void getAll(@NotNull Context ctx) {
+        String officeId = ctx.queryParam(OFFICE_MASK);
+        String locationId = ctx.queryParam(ID_MASK);
+        String unitSystem = ctx.queryParamAsClass(UNIT_SYSTEM, String.class).getOrDefault(UnitSystem.EN.value());
+        Instant minDate = parseInstant(ctx.queryParam(MIN_DATE));
+        Instant maxDate = parseInstant(ctx.queryParam(MAX_DATE));
+        String minNum = ctx.queryParam(MIN_NUMBER);
+        String maxNum = ctx.queryParam(MAX_NUMBER);
+        Number minHeight = null;
+        Number maxHeight = null;
+        Number minFlow = null;
```

Comment:
Are these not fields to be used as filters? Is there a reason they aren't available as input parameters?

Reply:
We aren't supporting these fields currently for filtering via the GET endpoint input parameters. I separated them out here in case that ever becomes something supported in the future.